merkle-tree/indexed/src/errors.rs

```diff
@@ -30,14 +35,16 @@ pub enum IndexedMerkleTreeError {
 impl From<IndexedMerkleTreeError> for u32 {
     fn from(e: IndexedMerkleTreeError) -> u32 {
         match e {
-            IndexedMerkleTreeError::IntegerOverflow => 3001,
-            IndexedMerkleTreeError::IndexHigherThanMax => 3002,
-            IndexedMerkleTreeError::LowElementNotFound => 3003,
-            IndexedMerkleTreeError::LowElementGreaterOrEqualToNewElement => 3004,
-            IndexedMerkleTreeError::NewElementGreaterOrEqualToNextElement => 3005,
+            IndexedMerkleTreeError::IntegerOverflow => 10001,
+            IndexedMerkleTreeError::IndexHigherThanMax => 10002,
+            IndexedMerkleTreeError::LowElementNotFound => 10003,
+            IndexedMerkleTreeError::LowElementGreaterOrEqualToNewElement => 10004,
+            IndexedMerkleTreeError::NewElementGreaterOrEqualToNextElement => 10005,
+            IndexedMerkleTreeError::ChangelogBufferSize(_, _) => 10006,
```

Comment:
is there a convention for error types like this?

Reply:
I know that anchor uses 6000

Reply:
Not really. I just tried to pick numbers not taken by Anchor

Reply:
> Not really. I just tried to pick numbers not taken by Anchor

But I've just realized I did a quite bad job here. Not in this PR, 10k is fine. But Anchor uses 100-5000:

https://docs.rs/anchor-lang/latest/anchor_lang/error/enum.ErrorCode.html
https://www.anchor-lang.com/docs/errors

We should make sure we take only ranges starting from 6k or 7k. Let's do it in a separate PR?

Reply:
one idea that crossed my mind is to introduce levels:
- hash-set: 7000ers
- Merkle-tree: 8000ers
- Indexed: 9000ers
- account-compression: 10000ers
- compressed-pda: 11000ers
- compressed-token: 12000ers

different idea:
- hash-set: 7000-7100
- Merkle-tree: 7100-7200
- Indexed: 7200-7300
- account-compression: 8000ers
- compressed-pda: 9000ers
- compressed-token: 10000ers